### PR TITLE
add support for reconciling a series that is downloading

### PR DESCRIPTION
**Pull request by kdwils**
I haven't actually tested this yet

----

> [!IMPORTANT]
> Add support for reconciling downloading series, including handling of episodes and seasons in various states, with updates to database schema and tests.
> 
>   - **Behavior**:
>     - Adds `ReconcileDownloadingSeries` to `series_reconcile.go` to handle series with downloading episodes.
>     - Introduces `reconcileDownloadingSeason` and `reconcileDownloadingEpisode` to manage downloading states.
>     - Updates `processIndividualEpisodeDownload` to handle completed downloads.
>     - Modifies `addEpisodeFileToLibrary` to handle existing files.
>   - **Database**:
>     - Updates `CreateSeriesMetadata`, `CreateSeasonMetadata`, and `CreateEpisodeMetadata` in `series.go` to handle upserts.
>     - Adds unique indexes for transitions in `schema.sql`.
>   - **Tests**:
>     - Adds tests for `ReconcileDownloadingSeries` and related functions in `series_reconcile_test.go`.
>     - Updates existing tests to cover new behavior and edge cases.
>   - **Misc**:
>     - Renames `reconcile.go` to `movie_reconcile.go` and updates references.
>     - Fixes minor issues in `sqlite.go` and `series.go` related to transaction handling.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=kasuboski%2Fmediaz&utm_source=github&utm_medium=referral)<sup> for 14efcf2b01577b6c9e2b00d8eeb0a9d1782b3ac3. You can [customize](https://app.ellipsis.dev/kasuboski/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: pkg/manager/series_reconcile.go [hunk 1]
```diff
@@ -896,3 +844,338 @@ func (m MediaManager) ReconcileCompletedSeasons(ctx context.Context) error {
...
+		}
+
+		log.Debug("linked episode to file", zap.Int32("episode_id", episode.ID), zap.String("path", ef.RelativePath))
+	}
+
+	return nil
+}
+
+// matchEpisodeFileToEpisodes attempts to match an episode file to specific episodes based on filename patterns
+func (m MediaManager) matchEpisodeFileToEpisodes(ctx context.Context, filePath string, seasonEpisodes []*storage.Episode) []*storage.Episode {
```
Comment (kdwils) [inline pkg/manager/series_reconcile.go:1149]: I think this matching might eventually need enhanced. When we download an entire pack, we need to match the files included in the pack to their respective episodes in the database

File: pkg/manager/series_reconcile.go [hunk 2]
```diff
@@ -1287,3 +1235,338 @@ func (m MediaManager) ReconcileCompletedSeasons(ctx context.Context) error {
...
+	log := logger.FromCtx(ctx)
+	fileName := filepath.Base(filePath)
+
+	// Extract episode numbers from filename using regex patterns
+	matches := episodeNumberPattern.FindStringSubmatch(fileName)
```
[the remark this answers is not in the document]
Comment (kdwils) [reply pkg/manager/series_reconcile.go:1545]: ur dumb

File: pkg/manager/series_reconcile.go [hunk 3]
```diff
@@ -896,3 +844,338 @@ func (m MediaManager) ReconcileCompletedSeasons(ctx context.Context) error {
...
+		Size:             ef.Size,
+		RelativePath:     &ef.RelativePath,
+		OriginalFilePath: &filePath,
+	})
+	if err != nil {
+		log.Error("failed to create episode file record", zap.Error(err))
+		return err
+	}
+
+	// Link all episodes to this file (for season packs, multiple episodes share one file)
```
[the remark this answers is not in the document]
Comment (kdwils) [reply pkg/manager/series_reconcile.go]: supposedly this is fixed
